src/HttpInstrumentation.php

```diff
@@ -62,13 +62,14 @@ public static function register(CachedInstrumentation $instrumentation): void
 
                 return [$request];
             },
-            post: static function (Application $app, array $params, ?Response $response, ?Throwable $exception) {
+            post: static function (Application $app, array $params, ?Response $response, ?Throwable $exception) use($request) {
                 $scope = Context::storage()->scope();
                 if (!$scope) {
                     return;
                 }
                 $scope->detach();
                 $span = Span::fromContext($scope->context());
+                $span->updateName(self::getSpanName($app, $request));
```

Comment:
does this have to happen here because the routing is not loaded in the `pre`?

Reply:
Yeah, the route does not get resolved until here. It's the same way it's done in the laravel side https://github.com/open-telemetry/opentelemetry-php-contrib/blob/6be232d7d5b3a0b11b7474d499a760a7ddaf616d/src/Instrumentation/Laravel/src/Hooks/Illuminate/Contracts/Http/Kernel.php#L78-L88 but have to do a bit more work because of lumen